### PR TITLE
Add test funciton to compare files in compute nodes to catch bugs in …

### PR DESCRIPTION
**Pull request by lubennikovaav**
…SLRU replay.

Compare files in existing compute node's pgdata with fresh basebackup at the same lsn. We expect that content is identical, except tmp files
Use it after some tests.

### CODE REVIEW

File: test_runner/fixtures/zenith_fixtures.py [hunk 1]
```diff
@@ -517,6 +518,23 @@ def __exit__(self, exc_type, exc, tb):
         self.stop()
 
 
+    def list_files_to_compare(self):
+        pgdata_files = []
+        for root, __file__, filenames in os.walk(self.pgdata_dir):
```
Comment (LizardWizzard) [inline test_runner/fixtures/zenith_fixtures.py:523]: `__file__` has a special meaning in python (all dunderscore methods or variables typically carry some additional meaning). Probably in that case you can use `dirnames` if you do not need the value you can use `_` like in Rust

Also you can skip unwanted directories by removing items from this `dirnames` variable. See an example from our performance tests: https://github.com/zenithdb/zenith/blob/main/test_runner/performance/test_perf_pgbench.py#L15

**Review by LizardWizzard — APPROVED** [review]
Looks good to me. What do you think, will it be beneficial to add this check to all tests by default? (we can exclude it from irrelevant ones, or use something like `@pytest.mark.verify_pg_files_compat` for enabling this check)

File: test_runner/fixtures/zenith_fixtures.py [hunk 2]
```diff
@@ -916,3 +934,41 @@ def create(self, tenant_id: Optional[str] = None):
...
+    # stop postgres to ensure that files won't change
+    pg.stop()
+
+    # list files we're going to compare
+    pgdata_files = pg.list_files_to_compare()
+
+    # create new branch, but don't start postgres
+    # We only need 'basebackup' result here.
+    zenith_cli.run(
+        ["branch", "new", pg.branch + "@" + lsn])
```
Comment (kelvich) [inline test_runner/fixtures/zenith_fixtures.py:951]: smth like "test_check_nonrel" would be clearer

File: test_runner/fixtures/zenith_fixtures.py [hunk 3]
```diff
@@ -916,3 +934,41 @@ def create(self, tenant_id: Optional[str] = None):
...
+    # compare content of the files
+    # filecmp returns (match, mismatch, error) lists
+    # We've already filtered all mismatching files in list_files_to_compare(),
+    # so here expect that the content is identical
+    (match, mismatch, error) = filecmp.cmpfiles(pg.pgdata_dir,
+                                                pg2.pgdata_dir,
+                                                pgdata_files,
+                                                shallow=False)
+    print('filecmp result mismatch and error lists:')
+    print(mismatch)
```
Comment (kelvich) [inline test_runner/fixtures/zenith_fixtures.py:972]: Let's maybe call diff on mismatched files? Smth like that:

```
xxd -b test_output/test_pg_regress/repo/pgdatadirs/tenants/b947bda26b5f33fe571daf2587ec4750/test_pg_regress/pg_xact/0000 > f1
xxd -b test_output/test_pg_regress/repo/pgdatadirs/tenants/b947bda26b5f33fe571daf2587ec4750/new/pg_xact/0000 > f2
> diff f1 f2                                                                                                       
204c204
< 000004c2: 01010101 01010101 01010101 01010101 01010101 01010101  UUUUUU
---
> 000004c2: 01010101 01010101 01010101 01010101 01010101 01010111  UUUUUW
262,265c262,265
< 0000061e: 01011001 01010101 01010101 01010101 10010101 01010101  YUUU.U
< 00000624: 01010101 01010101 01010101 01010101 01101001 01010101  UUUUiU
< 0000062a: 10011001 10101010 01010110 01010101 01010101 01010101  ..VUUU
< 00000630: 01010110 01010101 01101001 01010101 10100101 01010110  VUiU.V
---
> 0000061e: 01011001 01010101 01010101 01010101 10110101 01010101  YUUU.U
> 00000624: 01010101 01010101 01010101 01010101 01101011 11010111  UUUUk.
> 0000062a: 10011011 10101010 01110110 01010101 01010101 01010101  ..vUUU
> 00000630: 01011110 01010101 11101001 01110111 10100101 01011110  ^U.w.^
498,499c498,499
< 00000ba6: 01010101 10010101 01010101 01010101 01010101 10101001  U.UUU.
< 00000bac: 01010101 01010101 01010101 01010101 01010101 01100101  UUUUUe
---
> 00000ba6: 01010111 10010101 01011101 01010101 01010101 10101001  W.]UU.
> 00000bac: 01010101 01010101 01010101 01011101 01010101 01100101  UUU]Ue
691c691
< 0000102c: 01100101 01010110 01010101 01010101 01010101 01010101  eVUUUU
---
> 0000102c: 01100111 01010110 01010101 01010101 01010101 01010101  gVUUUU
```

File: test_runner/fixtures/zenith_fixtures.py [hunk 4]
```diff
@@ -971,4 +972,14 @@ def check_restored_datadir_content(zenith_cli, pg, lsn, postgres: PostgresFactor
     print('filecmp result mismatch and error lists:')
     print(mismatch)
     print(error)
+
+    for f in mismatch:
+
+        f1 = os.path.join(pg.pgdata_dir, f)
+        f2 = os.path.join(pg2.pgdata_dir, f)
+
+        cmd = ['diff', f1, f2]
```
Comment (kelvich) [inline test_runner/fixtures/zenith_fixtures.py:981]: ? that would print `Binary files x1 x2 differ`

**Pull request merged**